frontend/controller/actions/group-kv.js

```diff
@@ -5,8 +5,8 @@ import { KV_QUEUE } from '~/frontend/utils/events.js'
 
 export default (sbp('sbp/selectors/register', {
   'gi.actions/group/kv/updateLastLoggedIn': async ({ contractID }: { contractID: string }) => {
-    const { ourIdentityContractId } = sbp('state/vuex/getters')
-    if (!ourIdentityContractId) {
+    const identityContractID = sbp('chelonia/rootState').loggedIn?.identityContractID
```

Comment:
I assume this change is because for some reason we're not allowed to use root getters in the service worker? 🤔 

(Maybe there's a way to enable that...?)

Reply:
Correct. There isn't a way to enable that without code duplication (of the bundle kind, not of the source kind). However, global getters are a bad idea for contracts because they make it difficult or impossible to enforce containment, which is a requirement for any meaningful sandboxing, or to audit what contracts do. Furthermore, getters will likely need to be asynchronous if contracts are to be sandboxed at a later time.

Getters are fine to use and have in the application itself, but I wouldn't say they are well-suited to the service worker environment, where they just add complexity.

In a similar vein, `sbp('chelonia/rootState')` is also not well suited for the SW environment, for the same reasons. However, this could later be replaced (in contracts) with code that represents the running environment. For example, when a contract is loaded, Chelonia could give it certain 'globals' with state (e.g., login information) that contracts can access and would work similarly to accessing the root state / getters today.